lib/engine/route.rb

```diff
@@ -235,11 +235,13 @@ def connection_hexes
       connections.map(&:id)
     end
 
-    def find_connections(connections_a, connections_b)
+    def find_connections(connections_a, connections_b, other_paths)
       connections_a.each do |a|
+        next if (a.paths & other_paths).any?
+
         connections_b.each do |b|
           middle = (a.nodes & b.nodes)
-          next if middle.empty?
+          next if middle.empty? || (b.paths & other_paths).any? || (b.paths & a.paths).any?
```

Comment:
maybe we should move these & other_paths to a select+ that way it's N + M as opposed to N X M?

Reply:
Can you clarify? I don't understand.

Reply:
I could build a new version of other_paths inside the outer loop that is other_paths + a.paths. Is that what you mean?

Reply:
well since you do a next if b.paths & other_paths.any? in this inner loop, you could instead do connections_b.select(...) outside and use that instead of connections_b

Reply:
Gotcha